rest/handler.go

```diff
@@ -58,37 +59,51 @@ func TxHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	txMessageBytes, err := hex.DecodeString(tx.Message)
 	if err != nil {
-		WriteErrorResponse(w, http.StatusBadRequest, "Cannot decode message")
+		WriteErrorResponse(w, http.StatusBadRequest, err.Error())
```

Comment:
Sorry I was not saying clearly in the call. But let me send another pr to illustrate the idea.

Reply:
#148 is the pr.

Reply:
Ah I see what you mean, but what do you think of retaining control of what error code we want to send over an error, the current implementation was using `BadRequest` everywhere and there was a fix coming for custom status code with error since that is required by wallets and explorers to detect what went wrong.

Reply:
Alright, if that's the case, the current approach looks fine.